rfcs/0003-rfc-process/README.md

```diff
@@ -0,0 +1,119 @@
+
+1. The author clones this repository, creates a new directory in the `drafts`
+   directory with `X`s in place of the RFC number, (e.g.,
+   `drafts/XXXX-new_proposal/`), and copies the root-level `template.md` file
+   there as `README.md`.
+2. The author modifies the new `README.md` file to contain their newly proposed
+   RFC.
+3. If supplemental materials are needed (this should be generally rare), they
+   can go into the new directory alongside the `README.md` file.
+4. The author opens a pull request. The title of the pull request should be the
+   RFC title, and the description should be the overview text from the RFC. The
+   description should end with a hyperlink reading `rendered`, linking to the
+   RFC file itself.
+5. Reviewers provide feedback, engage in discussion, etc., just as in a normal
+   GitHub-based code review.
+6. The RFC becomes "settled". Settlement involves one of several paths of
+   action, detailed next.
+
+### Settling an RFC
+
+RFCs can be settled in several ways.
+
+**Acceptance.** An RFC can be accepted as written (which includes mutually
+agreed-upon edits).
+
+1. Two reviewers provide a review approval.
+2. The author provides their "final cut" (that is, their intention to make no
+   further updates to the document).
+3. A reviewer or the author determines the RFC number (by ascertaining the
+   highest number in use, and incrementing that).
+4. That person renames the draft directory using the four-digit number in
+   place of the `XXXX` draft prefix.
+5. The person moves the draft directory to the `accepted` top level directory,
+   and merges the pull request.
+
+**Rejection.** The RFC can be rejected if it does not fit into the larger aims
+of the project.
+
+1. Discussion ends with a determination that the RFC should be rejected.
+2. A reviewer or the author determines the RFC number (by ascertaining the
+   highest number in use, and incrementing that).
+3. That person renames the draft directory using the four-digit number in place
+   of the `XXXX` draft prefix.
+4. The person moves the draft directory to the `rejected` top level directory,
+   and merges the pull request.
+
+Note that despite **rejecting** the RFC, this procedure **accepts** the pull
+request. This is necessary to preserve the history of ideas and information that
+flowed through the project. The RFC number must still be assigned to make
+reference to this RFC easier in the future.
+
+If rejection leads to an obvious successor RFC candidate, that new RFC can be
+worked on in the same pull request thread that moves the initial RFC to the
+`rejection` state.
+
+**Withdrawal.** The RFC can be withdrawn by the author.
+
+1. The author determines the appropriate RFC number and renames their draft
+   directory as in the cases above.
+2. The author moves the draft directory to the `withdrawn` top level directory
+   and informs the pull request thread of their intention to withdraw.
+3. A reviewer merges the pull request.
+
```

Comment:
It seems that in all of the cases here, an RFC number is always added. Is there a reason not to specify this number upon folder creation, rather than adding it in later? Is this just to reduce possible friction between simultaneous PRs?

Reply:
Yeah I think that's exactly why. It shouldn't be a frequent problem for us but I don't think it's a big deal to do it this way either, and it will forestall trouble the first time it does happen.